plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -658,10 +659,9 @@ private void addSplitPluginDependencies(String thisPlugin, MavenRunner.Config mc
                     System.out.println("Skipping implicit dep " + thisPlugin + " → " + plugin);
                     continue;
                 }
-                VersionNumber splitPoint = new VersionNumber(pieces[1]);
+                String splitPoint = pieces[1];
                 VersionNumber declaredMinimum = new VersionNumber(pieces[2]);
-                // TODO this should only happen if the tested core version is ≥ splitPoint
-                if (coreDep.compareTo(splitPoint) <= 0 && !pluginDeps.containsKey(plugin)) {
+                if (coreCoordinates.compareVersionTo(splitPoint) >= 0 && !pluginDeps.containsKey(plugin)) {
```

Comment:
It would be great to have an in-repo integration test system so that subtle changes like this could be proven using new test cases.